### PR TITLE
Documentation site

### PR DESCRIPTION
**Pull request by jthegedus**
# Summary

Implemented documentation site using [Docsify](https://docsify.js.org/#/) as mentioned in https://github.com/asdf-vm/asdf/issues/274

If you wish to have a look, clone the PR/or my fork and follow the instructions in the `docs/contributing-doc-site.md` guide.

:100: **This adds 0 extra tooling to the repo. It does not require any compilation. Just edit the markdown.** :100: 

I am happy to guide and review all future additions/edits/removals to the `docs/` folder. Just ping me :tada:

asdf-vm doc site sample
Ignore the bad compression of the gif.
Some things I want to point out:
* copy to clipboard
* search
* tabbed content (just markdown, super nice)
* all-plugins page renders that repos readme here

## Other Information
* Please squash-merge this.
* some more optimizations can be made. For instance, we may want the search functionality to not search the changelog :man_shrugging: 

These are my recommendation for next steps:

- [ ] toggle GitHub Pages using the `docs/` folder setting - https://docsify.js.org/#/deploy?id=github-pages
- [ ] set up redirect with gh-pages for `asdf-vm.com` (as I mentioned in #274 I have secured this domain)
- [ ] update main README to direct people to new doc site
- [ ] remove most content from README so it is more sparse. Good examples of this are https://github.com/prettier/prettier or https://github.com/docsifyjs/docsify/

### CODE REVIEW

**Comment by jthegedus** [comment]
I'm unsure who would most want to review this, sorry to ping you all.

**Comment by jthegedus** [comment]
I'm happy to merge this myself, but figured since this effects everyone I wanted someone else to at least look first

**Comment by Stratus3D** [comment]
One question this PR brings to mind is what is the proper format for shell commands? There are two styles I've seen. The first is with shell command lines starting with `$ `:

```shell
$ asdf current
ruby 2.6.0
```

Or with output lines commented out:

```shell
asdf current
# ruby 2.6.0
```

**Comment by jthegedus** [comment]
Should be good to go once CI passes

File: docs/core-manage-asdf-vm.md [hunk 1]
```diff
@@ -0,0 +1,123 @@
+## Install asdf-vm
+
+Clone only the latest branch:
+
+```shell
+git clone https://github.com/asdf-vm/asdf.git ~/.asdf --branch v0.6.2
```
[the remark this answers is not in the document]
Comment (Stratus3D) [reply docs/core-manage-asdf-vm.md]: All you should need to do is change the line that runs a sed command on the README and change it so it either runs on both files or just this one.

[the remark this answers is not in the document]
Comment (jthegedus) [reply docs/core-manage-asdf-vm.md]: Like so? https://github.com/asdf-vm/asdf/pull/422/commits/b644cf2131b26f57b0c4f7286d6ea19219e76b34

**Comment by Stratus3D** [comment]
Can I merge this?

File: release.sh [hunk 2]
```diff
@@ -100,6 +100,10 @@ fi
 sed -i.bak "s|^\\(git clone.*--branch \\).*$|\\1$new_tag_name|" README.md
 rm README.md.bak
 
+# Update version in docs/core-manage-asdf-vm.md
+sed -i.bak "s|^\\(git clone.*--branch \\).*$|\\1$new_tag_name|" docs/core-manage-asdf-vm.md
+rm docs/core-manage-asdf-vm.md.bak
```
Comment (Stratus3D) [inline release.sh]: Thanks! Looks great!

**Comment by Stratus3D** [comment]
Squash merged!

**Comment by Stratus3D** [comment]
@jthegedus thanks for all your work on this. Can you complete those other tasks on your own? Or do you need the repo to setup Github pages?

**Comment by jthegedus** [comment]
I believe I have the access to complete these additional tasks myself. Thanks for reviewing, I know it was a lot to get through :100:

**Comment by HashNuke** [comment]
Thank you @jthegedus and @Stratus3D ~!

**Pull request merged**